Sources/SwiftFusion/Geometry/Pose2.swift

```diff
@@ -1,13 +1,32 @@
 /// Pose2 class is the new Swift type for the SE(2) manifold of 2D Euclidean
 /// Poses
 public struct Pose2: Equatable, Differentiable, KeyPathIterable, TangentStandardBasis {
-  public var t_: Point2
-  public var rot_: Rot2
+  private var t_: Point2
+  private var rot_: Rot2
+
+  @differentiable public var t: Point2 { t_ }
+  @differentiable public var rot: Rot2 { rot_ }
+
+  public struct TangentVector:
+    Differentiable, AdditiveArithmetic, KeyPathIterable, VectorProtocol, ElementaryFunctions
+  {
+    public typealias VectorSpaceScalar = Double
+    public var t: Point2.TangentVector
```

Comment:
I see I have sort of unthinkingly used the `Point2.TangentVector` and `Rot2.TangentVector` types here to store the components of the tangent space. This doesn't really make sense, as you say. So I'll change this struct definition to define its own components. We don't get much from using `Point2.TangentVector` and `Rot2.TangentVector`.

Reply:
Actually, there are two concepts related to manifolds: the “true tangent space” which in this case is a set of special 3x3 matrices, and the “vector tangent space” that is isometric to it (via the `hat`  and `vee`  maps. In GTSAM we just use the latter, and so really the *only* thing we have to specify here - I think- is the dimensionality of the tangent space. Since this is *always* the same as the dimensionality of the manifold, even that should be derived from the Manifold itself. Less code = good ? :-)

Reply:
*isomorphic

Reply:
Yes, I can sort of imagine one day being able to specify a few properties of a Lie group (dimension, exp, special names for certain algebra basis vectors, ...?) and have a library (possibly with compiler support) synthesize all the required types and derivatives automatically.

Ideally this would be mostly just a library, maybe with a few hooks in the compiler so that the library can control the synthesis more than is possible today. (Right now Swift will synthesize everything you need when you manifold is a direct product of manifolds, but it can't do anything else.)

I will start thinking about how something like this would be possible. I think it'll be a while before I come up with anything because I'm still very new to Lie groups and I'll have to keep playing with them and reading up on how they work in GTSAM before I figure anything out.

In the meantime, I'd like to just explicitly declare the type of the tangent space.